buildvertical.targets

```diff
@@ -8,6 +8,22 @@
  <!-- Required for FindBestConfiguration task -->
  <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
 
+ <Target Name="AnnotateProjects"
+         BeforeTargets="BuildAllProjects"
+         Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
+   <!-- Clear ProjectWithConfiguration to prevent circular dependency -->
+   <ItemGroup><ProjectWithConfiguration Remove="@(ProjectWithConfiguration)" /></ItemGroup>
```

Comment:
I'm not sure I understand why this needed to be split out from the traversals? Can you help me understand? The only reall thing I noticed is the fact you are now mutating the Project item grou